x-pack/plugins/infra/public/utils/logs_overview_fetches.test.ts

```diff
@@ -80,8 +107,72 @@ describe('Logs UI Observability Homepage Functions', () => {
   });
 
   describe('getLogsOverviewDataFetcher()', () => {
-    it.skip('should work', async () => {
-      // Pending
+    beforeAll(() => {
+      mockedCallFetchLogSourceConfigurationAPI.mockResolvedValue({
+        data: {
+          configuration: {
+            logAlias: 'filebeat-*',
+            fields: { timestamp: '@timestamp', tiebreaker: '_doc' },
+          },
+        },
+      } as GetLogSourceConfigurationSuccessResponsePayload);
+    });
+
+    afterAll(() => {
+      mockedCallFetchLogSourceConfigurationAPI.mockReset();
+    });
+
+    it('should work', async () => {
+      const { mockedGetStartServices, dataResponder } = setup();
+
+      dataResponder.mockReturnValue({
+        rawResponse: {
+          aggregations: {
+            stats: {
+              buckets: [
+                {
+                  key: 'nginx',
+                  doc_count: 250, // Count is for 2 minutes
+                  series: {
+                    buckets: [
+                      // Counts are per 30 seconds
+                      { key: 1593430680000, doc_count: 25 },
+                      { key: 1593430710000, doc_count: 50 },
+                      { key: 1593430740000, doc_count: 75 },
+                      { key: 1593430770000, doc_count: 100 },
+                    ],
+                  },
+                },
+              ],
+            },
+          },
+        },
+      });
+
+      const fetchData = getLogsOverviewDataFetcher(mockedGetStartServices);
+      const response = await fetchData(DEFAULT_PARAMS);
+
+      expect(response).toMatchObject({
+        stats: {
+          nginx: {
+            label: 'nginx',
+            type: 'number',
+            // Rate is normalized to logs in one minute
+            value: 125,
+          },
+        },
+        series: {
+          nginx: {
+            coordinates: [
+              // Rates are normalized to logs in one minute
+              { x: 1593430680000, y: 50 },
+              { x: 1593430710000, y: 100 },
+              { x: 1593430740000, y: 150 },
+              { x: 1593430770000, y: 200 },
```

Comment:
Is that a valid way to scale? The buckets are still 30 seconds wide, so the invariant `entries = rate * time` doesn't hold anymore.

Reply:
Thanks for the video chat conversations. I agree that it's technically correct and I got confused by the 30 second bucket sizes.